include/Gaffer/MetadataAlgo.h

```diff
@@ -100,6 +100,16 @@ bool ancestorAffectedByChange( const Plug *plug, IECore::TypeId changedNodeTypeI
 /// Copies metadata from one target to another. The exclude pattern is used with StringAlgo::matchMultiple().
 void copy( const GraphComponent *from, GraphComponent *to, const StringAlgo::MatchPattern &exclude = "", bool persistentOnly = true, bool persistent = true );
 
+///
+enum Overwrite
+{
+	KeepExisting,
+	OverwriteExisting
+};
+
+/// Copy nodule and noodle color meta data from srcPlug to dstPlug
+void copyColors( const Gaffer::Plug *srcPlug , Gaffer::Plug *dstPlug, Overwrite overwrite );
```

Comment:
I think we should experiment with a few ideas which are clearer in C++ and idiomatic in python. I like the look of tagged_bool (or perhaps like the Named Parameter idiom)  might help - but I'm not sure when these API clean up tasks might get done.